server/routes/api/representatives.js

```diff
@@ -50,15 +50,32 @@ const ALLOWED_JURISDICTION_FILTERS = Object.keys(JURISDICTION_FILTER_MAP)
 // Endpoints
 
 // Get
-router.get('/:zipCode', async (req, res) => {
-  const { zipCode } = req.params
+router.get('/:searchText', async (req, res) => {
+  const { searchText } = req.params
   const { filter } = req.query
 
-  if (!zipCode.match(/^\d{5}(-\d{4})?$/)) {
+  /* check if valid ZIP code 
+    ^\d{5} - the start of the line has 5 digits
+    (-\d{4})?$ the end of the line has 4 digits preceded by a dash (optionally)
+  */
+  let isValidZIPcode = /^\d{5}(-\d{4})?$/.test(searchText)
+  /* check if valid street address
+     ^(\d{3,})\s? the start of the line can have 3 or more digits followed by a space
+     (\w{0,})\s the next part can have 2 or more letters followed by a space
+     ([a-zA-Z]{2,30})\s the next part must be an alphanetical string with 2-30 characters followed by a space
+     ([a-zA-Z]{2,15})\.?\s?  the next part must be an alphabetical string with 2-15 characters followed by a any digit (optionally) and/or space (optionally)
```

Comment:
This does not accept a digit as written, it accepts exactly `.` because the regex includes `\.` to escape the period.  Matching zero or one digit would be `\d?`. I'm not sure why this part is allowed to end in a single digit, though?

Reply:
@thyeggman 
A street address can have abbreviations like these: 101 W Main St. N. The dot makes sense to me. And actually, it accepts any single character after the limit. I have tested it with https://rubular.com/. For example, if you change the limit to 0-2 or 1 alphabetic characters and any single character after it` ^(\d{3,})\s?(\w{0,5})\s([a-zA-Z]{2,30})\s([a-zA-Z]{0,2}).?\s?(\w{0,5})$`, the results are as follows:

- 101 W Main S21 N will not have matches or be valid
- 101 W Main St1 N will be valid (any combination of two alphabetic characters plus any character will work, not only the dot).

Reply:
I have made a mistake and fixed it by removing the escaping character.